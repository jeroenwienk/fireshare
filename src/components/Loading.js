import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Loading extends Component {
  static propTypes = {
    error: PropTypes.bool,
    retry: PropTypes.func,
    pastDelay: PropTypes.bool
  };

  render() {
    const { error, retry, pastDelay } = this.props;

    if (error) {
      return (
        <div>
          Error! <button onClick={retry}>Retry</button>
        </div>
      );
    }

    if (pastDelay) {
      return <div>Loading...</div>;
    }

    return null;
  }
}

export default Loading;
