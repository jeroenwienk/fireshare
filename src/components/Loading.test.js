import React from 'react';
import renderer from 'react-test-renderer';
import Loading from './Loading';

describe('<Loading />', () => {
  test('it renders null with nothing', () => {
    const tree = renderer
      .create(<Loading error={false} pastDelay={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders with error', () => {
    const tree = renderer
      .create(<Loading error={true} pastDelay={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders with pastDelay', () => {
    const tree = renderer
      .create(<Loading error={false} pastDelay={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
