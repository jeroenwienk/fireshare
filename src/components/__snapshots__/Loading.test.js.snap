// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> it renders null with nothing 1`] = `null`;

exports[`<Loading /> it renders with error 1`] = `
<div>
  Error! 
  <button>
    Retry
  </button>
</div>
`;

exports[`<Loading /> it renders with pastDelay 1`] = `
<div>
  Loading...
</div>
`;
